### PR TITLE
Add more logs

### PR DESCRIPTION
**Pull request by Kyle-Eldridge**
## Overview

This PR will add more logs.

## Design Doc

None. This is a small PR (1 commit).

## Tests Ran

I still don't have the robot.

### CODE REVIEW

**Review by omegaladon — DISMISSED** [review]

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java [hunk 1]
```diff
@@ -60,15 +62,20 @@ public void execute(){
...
           driveDistance, visionDistance);
         System.out.printf("Difference: %.4f\nPercent difference: %.4f%%\n",
           visionDistance - driveDistance, (visionDistance - driveDistance) / driveDistance * 100);
       } 
+      if(Constants.kLogging){
+        LogManager.addDoubleArray("Vision/DistanceTest", new double[]{
+          driveDistance, visionDistance, 
+          visionDistance - driveDistance, (visionDistance - driveDistance) / driveDistance * 100
+        });
+      }
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/commands/vision/TestVisionDistance.java:78]: perhaps it is a good idea to log this in the drivetrain rather than in this spicific test. I know it would be longging in this test but I think if you are already printing it out it is prorobly good enough for what your are doing then again it dosen't really hurt to leave it here. 

TLDR; add this to drivetrain not just here.

**Review by PandaNinja10 — DISMISSED** [review]

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java [hunk 2]
```diff
@@ -60,15 +62,20 @@ public void execute(){
       // reset the timer
       m_endTimer.reset();
       // If kPrintDelay seconds have passed, print the data
-      // TODO: Consider using shuffleboard instead of printing
```
[the remark this answers is not in the document]
Comment (Kyle-Eldridge) [reply src/main/java/frc/robot/commands/vision/TestVisionDistance.java:63]: I added logging later that function, which I think uses Shuffleboard.

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java (hunk 1 above)
Comment (Kyle-Eldridge) [reply src/main/java/frc/robot/commands/vision/TestVisionDistance.java:78]: This is the distance from vision and encoders, and is specific to this test. I kept both the print statements and this because print statements are easier to access (DriverStation logs), but logging could be helpful to see all of the data. It also logs constantly instead of every second, which could give better data. I think the pose is already logged in drivetrain, so I'm not sure what this would do. It measures distance from the starting point in this test, and we don't really need distance from the actual starting point to wherever the robot is on the field. This test also disables vision, which we should not do on the actual robot during SFR.

**Comment by Kyle-Eldridge** [comment]
@PandaNinja10 I looked at your feedback and added the changes you mentioned.

File: src/main/java/frc/robot/subsystems/Module.java [hunk 3]
```diff
@@ -75,6 +87,12 @@ public void setDesiredState(SwerveModuleState desiredState, boolean isOpenLoop)
     m_desiredState = desiredState;
     setAngle(desiredState);
     setSpeed(desiredState, isOpenLoop);
+
+    if(Constants.kLogging){
+      LogManager.addDoubleArray("Swerve/Modules/"+m_moduleAbbr+"/DesiredState", new double[]{
+        m_moduleIndex, desiredState.speedMetersPerSecond, desiredState.angle.getDegrees()
```
Comment (AnthonyFurman) [inline src/main/java/frc/robot/subsystems/Module.java:93]: probably shouldn't log the module index

File: src/main/java/frc/robot/subsystems/Module.java [hunk 4]
```diff
@@ -87,6 +105,19 @@ private void setSpeed(SwerveModuleState desiredState, boolean isOpenLoop) {
...
+      LogManager.addDoubleArray("Swerve/Modules/"+m_moduleAbbr+"/DriveValues",
+        new double[]{
+          desiredState.speedMetersPerSecond, 
+          Conversions.falconToMPS(m_driveMotor.getSelectedSensorVelocity(), DriveConstants.kWheelCircumference,
+            DriveConstants.kDriveGearRatio),
+          Conversions.falconToMPS(m_driveMotor.getSelectedSensorVelocity(), DriveConstants.kWheelCircumference,
+            DriveConstants.kDriveGearRatio)-desiredState.speedMetersPerSecond,
+          m_driveMotor.getMotorOutputVoltage(), 
+          m_driveMotor.getStatorCurrent()
+        }
```
Comment (AnthonyFurman) [inline src/main/java/frc/robot/subsystems/Module.java:118]: don't log all this stuff in one array. Also, this is probably pinging the motors too much and will cause issues

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (PandaNinja10) [reply src/main/java/frc/robot/commands/vision/TestVisionDistance.java:63]: logging dose not use suffleboard

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java (hunk 1 above)
Comment (PandaNinja10) [reply src/main/java/frc/robot/commands/vision/TestVisionDistance.java:85]: I was more thinking the distance between the apirl tag and the camera or robot.

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kyle-Eldridge) [reply src/main/java/frc/robot/commands/vision/TestVisionDistance.java:63]: It uses Shuffleboard now, too.

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java [hunk 5]
```diff
@@ -28,11 +32,13 @@ public class TestVisionDistance extends CommandBase{
...
   private static final double kPrintDelay = 0.25;
 
-  public TestVisionDistance(double speed, Drivetrain drive, Vision vision){
+  public TestVisionDistance(double speed, Drivetrain drive, Vision vision, ShuffleboardTab visionTab){
     addRequirements(drive);
     m_drive = drive;
     m_speed = speed;
     m_vision = vision;
+    m_outputString = "Run the vision distance test";
+    visionTab.addString("Distance Test Results", ()->getOutputString());
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/commands/vision/TestVisionDistance.java:41]: It would be better to put number on shuffle board rather than a string wich is difficult to read in the shuffl eboar formant and loses features such as graphing spcific to numbers

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java [hunk 6]
```diff
@@ -60,15 +67,22 @@ public void execute(){
...
-        System.out.printf("Difference: %.4f\nPercent difference: %.4f%%\n",
-          visionDistance - driveDistance, (visionDistance - driveDistance) / driveDistance * 100);
-      } 
+        System.out.println(m_outputString);
+      }
+      if(Constants.kLogging){
+        LogManager.addDoubleArray("Vision/DistanceTest", new double[]{
+          driveDistance, visionDistance, 
+          visionDistance - driveDistance, (visionDistance - driveDistance) / driveDistance * 100
+        });
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/commands/vision/TestVisionDistance.java:84]: It may be benificial to have seprate log entries rather than useing a double so it is clear wich messurement is which

Comment (Kyle-Eldridge) [reply src/main/java/frc/robot/commands/vision/TestVisionDistance.java:83]: If I did that, would it keep all 4 numbers together or put them on the same row in a table? They aren't useful unless they can be compared to each other.

File: src/main/java/frc/robot/subsystems/Module.java [hunk 7]
```diff
@@ -95,16 +130,36 @@ private void setAngle(SwerveModuleState desiredState) {
...
+          Conversions.falconToDegrees(m_angleMotor.getSelectedSensorVelocity(), 
+            DriveConstants.kAngleGearRatio)
+        }
+      );
+      LogManager.addDoubleArray("Swerve/Modules/"+m_moduleAbbr+"/SteerPower",
+        new double[]{
+          m_angleMotor.getMotorOutputVoltage(), 
+          m_angleMotor.getStatorCurrent(),
+        }
+      );
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/subsystems/Module.java:151]: desisered angle and position are already logged in drivetrain. The other things being logges should not be in arrays but individual logs so it is clear to someone looking at the logs what they represent

File: src/main/java/frc/robot/subsystems/Module.java [hunk 8]
```diff
@@ -1,5 +1,17 @@
 package frc.robot.subsystems;
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/subsystems/Module.java:1]: I noticed that all the loggs go Swerve/ID/item. I think it would be better to have it Swerve/item/ID becasue you are more likely to want to look at items of the same type rathe rather than all items of a spcific module

**Comment by PandaNinja10** [comment]
> @PandaNinja10 It says you requested changes, but I can't see what changes you requested. Is there somewhere I can see the specific changes, or does it just reopen the PR?

I think you resolved all the changes I requested at the time IDK why didn't reslove. I will probobly avoid the github feature in the future to avoid confusion

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java (hunk 6 above)
Comment (PandaNinja10) [reply src/main/java/frc/robot/commands/vision/TestVisionDistance.java:83]: you can indicudualy drag them in to compare them in advantage scope. But it is really dificult to compare things that you aren't sure what they are.

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java [hunk 9]
```diff
@@ -28,11 +32,13 @@ public class TestVisionDistance extends CommandBase{
...
   private static final double kPrintDelay = 0.25;
 
-  public TestVisionDistance(double speed, Drivetrain drive, Vision vision){
+  public TestVisionDistance(double speed, Drivetrain drive, Vision vision, ShuffleboardTab visionTab){
     addRequirements(drive);
     m_drive = drive;
     m_speed = speed;
     m_vision = vision;
+    m_outputData = new double[4];
+    visionTab.addDoubleArray("Distance Test Results", ()->getOutputData());
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/commands/vision/TestVisionDistance.java:41]: again please label what you outputing rather than just outputing an array and assuming people know what and the order of the output

**Comment by Kyle-Eldridge** [comment]
@AnthonyFurman @PandaNinja10 @richtan This I think this PR is done. Can you merge it?

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java [hunk 10]
```diff
@@ -28,11 +32,14 @@ public class TestVisionDistance extends CommandBase{
...
   private static final double kPrintDelay = 0.25;
 
-  public TestVisionDistance(double speed, Drivetrain drive, Vision vision){
+  public TestVisionDistance(double speed, Drivetrain drive, Vision vision, ShuffleboardTab visionTab){
     addRequirements(drive);
     m_drive = drive;
     m_speed = speed;
     m_vision = vision;
+    m_outputData = new double[4];
+    visionTab.addDoubleArray("Distance Test Results\n(drive, vision, difference, % difference)",
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/commands/vision/TestVisionDistance.java:41]: posible issue, if 2 instances of the command are created the robot will spit out an error becuse the shuffleboard title is already in use.

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java [hunk 11]
```diff
@@ -28,11 +32,14 @@ public class TestVisionDistance extends CommandBase{
...
 
-  public TestVisionDistance(double speed, Drivetrain drive, Vision vision){
+  public TestVisionDistance(double speed, Drivetrain drive, Vision vision, ShuffleboardTab visionTab){
     addRequirements(drive);
     m_drive = drive;
     m_speed = speed;
     m_vision = vision;
+    m_outputData = new double[4];
+    visionTab.addDoubleArray("Distance Test Results\n(drive, vision, difference, % difference)", 
+      ()->getOutputData());
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/commands/vision/TestVisionDistance.java:42]: please don't output arrays to shuffleboard

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 12]
```diff
@@ -406,9 +406,13 @@ public void updateOdometry() {
...
           estimatedPose.timestampSeconds,
-          m_chargeStationVision ? VisionConstants.kChargeStationVisionPoseStdDevs.plus(
-            currentEstimatedPoseTranslation.getDistance(closestTagPoseTranslation) * VisionConstants.kVisionPoseStdDevFactor
-          ) : VisionConstants.kBaseVisionPoseStdDevs
+          m_chargeStationVision ? VisionConstants.kChargeStationVisionPoseStdDevs :
+            VisionConstants.kBaseVisionPoseStdDevs.plus(
+              currentEstimatedPoseTranslation.getDistance(closestTagPoseTranslation) * VisionConstants.kVisionPoseStdDevFactor
+            )
+        );
+        LogManager.addDouble("Vision/ClosestTag",
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:414]: maybe clarify distance

File: src/main/java/frc/robot/controls/Operator.java [hunk 13]
```diff
@@ -10,11 +10,13 @@
 import frc.robot.commands.scoring.PositionIntake.Position;
 import frc.robot.commands.scoring.elevator.CalibrateElevator;
 import frc.robot.commands.scoring.intake.IntakeGamePiece;
+import frc.robot.constants.Constants;
 import frc.robot.commands.scoring.intake.OuttakeGamePiece;
 import frc.robot.commands.scoring.wrist.RotateWrist;
 import frc.robot.constants.OIConstants;
 import frc.robot.constants.WristConstants;
 import frc.robot.subsystems.Elevator;
+import frc.robot.util.LogManager;
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/controls/Operator.java:19]: random imports not needed

File: src/main/java/frc/robot/util/Vision.java [hunk 14]
```diff
@@ -197,12 +206,27 @@ public Optional<EstimatedRobotPose> getEstimatedPose(Pose2d referencePose) {
   }
 
   public void addTestCommands(ShuffleboardTab testTab, GenericEntry testEntry, Drivetrain drive){
-    testTab.add("Calculate vision std devs", new CalculateStdDevs(1000, drive, this).beforeStarting(new WaitCommand(5)));
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/util/Vision.java:200]: why is this removed

File: src/main/java/frc/robot/commands/vision/TestVisionDistance.java [hunk 15]
```diff
@@ -102,4 +117,11 @@ public boolean isFinished() {
     return m_endTimer.hasElapsed(kEndDelay);
   }
 
+  public double getDriveDistance() {
+    return m_driveDistance;
+  }
+
+  public double getVisionDistance() {
+    return m_visionDistance;
+  }
```
Comment (PandaNinja10) [inline src/main/java/frc/robot/commands/vision/TestVisionDistance.java:126]: do you ever use the getters or plan to use the getters

**Review by PandaNinja10 — APPROVED** [review]
